File: backend/src/main/java/com/lerneon/backend/models/payload/request/DepartmentRequest.java
package com.lerneon.backend.models.payload.request;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class DepartmentRequest {
    @NotBlank(message = "Department name is required")
    @Size(max = 100, message = "Department name must not exceed 100 characters")
    private String name;

    private String description;

    @NotNull(message = "Company ID is required")
    private Integer companyId;
}

File: client/types/payload/request/department.request.ts
import { z } from "zod/v3";

export const DepartmentRequest = z.object({
  name: z
    .string()
    .min(1, "Department name is required")
    .max(100, "Department name cannot exceed 100 characters"),
  description: z.string().min(1, "Department name is required"),
});

export type DepartmentRequest = z.infer<typeof DepartmentRequest>;
